pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,150 @@
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace(s)
+func (d *ChainDescriber) MakeGraph(namespaces []string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{}
+	for _, namespace := range namespaces {
+		glog.V(4).Infof("Loading build configurations from %q", namespace)
+		loaders = append(loaders, &bcLoader{namespace: namespace, lister: d.Interface})
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+
+	return g, nil
+}
+
+// Describe returns the output of the graph starting from the provided
+// image stream tag (name:tag)
+func (d *ChainDescriber) Describe(namespace, name, tag string) (string, error) {
+	g, err := d.MakeGraph(d.namespaces)
+	if err != nil {
+		return "", err
+	}
+	fmt.Println("Initial graph edges length:", len(g.Edges()))
+
+	partitioned, err := partition(g, namespace, name, tag)
+	if err != nil {
+		return "", err
+	}
+	fmt.Println("Final graph edges length:", len(partitioned.Edges()))
+
+	switch d.output {
+	case "dot":
+		data, err := dot.Marshal(partitioned, name, "", "  ", false)
+		if err != nil {
+			return "", err
+		}
+		return string(data), nil
+	case "plain":
+		return partitioned.String(), nil
+	}
+	return "", fmt.Errorf("unknown specified format %q", d.output)
+}
+
+// partition the provided graph down to a subgraph starting from the given
+// image stream tag
+func partition(g osgraph.Graph, namespace, name, tag string) (osgraph.Graph, error) {
+	// Retrieve the imageStreamTag node of interest
+	ist := g.Find(imagegraph.ImageStreamTagNodeName(imagegraph.MakeImageStreamTagObjectMeta(namespace, name, tag)))
+	if ist == nil {
+		return "", NotFoundErr(fmt.Sprintf("%s:%s", name, tag))
+	}
+
+	// Filter out all but BuildConfig and ImageStreamTag nodes
+	nodeFn := osgraph.NodesOfKind(buildgraph.BuildConfigNodeKind, imagegraph.ImageStreamTagNodeKind)
+	// Filter out 1) all but BuildInputImage and BuildOutput edges and
+	// 2) inbound edges to the image stream tag of interest
+	allowedKinds := util.NewStringSet(buildedges.BuildInputImageEdgeKind, buildedges.BuildOutputEdgeKind)
+	var edgeFn osgraph.EdgeFunc = func(g osgraph.Interface, head, tail graph.Node, edgeKinds util.StringSet) bool {
+		return allowedKinds.HasAny(edgeKinds.List()...) && ist != tail
+	}
+
+	// The only reason for creating this subgraph is to clip out any
+	// inbound edges to the image stream tag of interest
+	sub := g.Subgraph(nodeFn, edgeFn)
+	fmt.Println("First subgraph edges length:", len(sub.Edges()))
+	// Add a reverse edge for every existing edge on the graph resulting to
```

Comment:
The rebase brought in some new options.  If the digraph can masquerade as a undirected graph, you want to try this? https://godoc.org/github.com/gonum/graph/topo#ConnectedComponents